hs_communities/views/communities.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import absolute_import
+
+import json
+
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.models import Group, User
+from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.utils.html import mark_safe, escapejs
+from django.views.generic import TemplateView
+
+from hs_access_control.management.utilities import community_from_name_or_id
+from hs_access_control.models.community import Community
+from hs_communities.models import Topic
+
+
+class CollaborateView(TemplateView):
+    template_name = 'pages/collaborate.html'
+
+
+class CommunityView(TemplateView):
+    template_name = 'hs_communities/community.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(CommunityView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        # user_id = User.objects.get(pk=self.request.user.id)
+        grpfilter = self.request.GET.get('grp')
+
+        community_resources = community_from_name_or_id("CZO National Community").public_resources
+        groups = []
+        for c in community_resources:
+            if not any(str(c.group_id) == g.get('id') for g in groups):  # if the group id is not already present in the list
+                if c.group_name != "CZO National":  # The National Group is used to establish the entire Community
+                    res_count = len([r for r in community_resources if r.group_name == c.group_name])
+                    groups.append({'id': str(c.group_id), 'name': str(c.group_name), 'res_count': str(res_count)})
+
+        groups = sorted(groups, key=lambda key: key['name'])
+        return {
+            'community_resources': community_resources,
+            'groups': groups,
+            'grpfilter': grpfilter,
+        }
+
+
+class FindCommunitiesView(TemplateView):
+    template_name = 'hs_communities/find-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(FindCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+
+        return {
+            'communities_list': Community.objects.all()
+        }
+
+
+@method_decorator(login_required, name='dispatch')
+class MyCommunitiesView(TemplateView):
+    template_name = 'hs_communities/my-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(MyCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def group_to_community(self, grp, communities):
```

Comment:
SHouldn't this be done in a database call querying communities by the group?

Reply:
I'm not seeing it, this method just returns the community tied to a group, right?  Instead of looping through all member_groups in python, just add a filter to member_groups and pass in the grp.id.

Reply:
Addressed in 1525edfbdd129b348abdac49b6de34156c2608ed